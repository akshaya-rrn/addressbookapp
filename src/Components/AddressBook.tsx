import { Alert, AlertTitle, Box, Typography } from '@mui/material';
import { useEffect, useState } from 'react';
import { GetUsers } from '../services/users';
import { UserResponseResult } from '../types';
import { UsersCard } from './UserCard';


const ErrorComponent = () => {
    return <Alert severity="error">
    <AlertTitle>Invalid Request</AlertTitle>
        There is something wrong with the request â€” <strong>Please try again!</strong>
    </Alert>
};

function AddressBook() {
  const [users, setUsers] = useState<UserResponseResult[]>([]);
  const [showError, setError] = useState<boolean>(false);

  useEffect(() => {
    GetUsers().then((response) => {
    setError(false);
      setUsers(response);
    }).catch((err) => {
        setError(true);
    });
  }, []);


  return (
    <div>
      <Typography variant="h3" component={'h3'} alignContent="center">Address Book</Typography>
      <br/>
      <br/>
      { showError ? <ErrorComponent/> : null}
      { users.length > 0 ?  (
        <Box>
            <UsersCard usersData={users}/>
        </Box>) : null}
    </div>
  );
}

export default AddressBook;
