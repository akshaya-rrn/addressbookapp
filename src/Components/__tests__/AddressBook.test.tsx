import renderer from 'react-test-renderer';
import AddressBook from '../AddressBook';


describe('Address Book Component', () => {

  test('renders correctly', () => {
  const tree = renderer.create(<AddressBook/>).toJSON();
  expect(tree).toMatchSnapshot();
});

})

