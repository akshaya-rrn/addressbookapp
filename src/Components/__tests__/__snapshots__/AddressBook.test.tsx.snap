// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address Book Component renders correctly 1`] = `
<div>
  <h3
    className="MuiTypography-root MuiTypography-h3 css-1j19yin-MuiTypography-root"
  >
    Address Book
  </h3>
  <br />
  <br />
</div>
`;
